moveit_core/robot_state/src/attached_body.cpp

```diff
@@ -43,31 +43,45 @@ namespace moveit
 {
 namespace core
 {
-AttachedBody::AttachedBody(const LinkModel* parent_link_model, const std::string& id,
+AttachedBody::AttachedBody(const LinkModel* parent_link_model, const std::string& id, const Eigen::Isometry3d& pose,
                            const std::vector<shapes::ShapeConstPtr>& shapes,
-                           const EigenSTL::vector_Isometry3d& attach_trans, const std::set<std::string>& touch_links,
+                           const EigenSTL::vector_Isometry3d& shape_poses, const std::set<std::string>& touch_links,
                            const trajectory_msgs::JointTrajectory& detach_posture,
                            const FixedTransformsMap& subframe_poses)
   : parent_link_model_(parent_link_model)
+  , pose_(pose)
   , id_(id)
   , shapes_(shapes)
-  , attach_trans_(attach_trans)
+  , shape_poses_(shape_poses)
   , touch_links_(touch_links)
   , detach_posture_(detach_posture)
   , subframe_poses_(subframe_poses)
   , global_subframe_poses_(subframe_poses)
 {
-  for (const auto& t : attach_trans_)
+  ASSERT_ISOMETRY(pose)  // unsanitized input, could contain a non-isometry
+  for (const auto& t : shape_poses_)
   {
     ASSERT_ISOMETRY(t)  // unsanitized input, could contain a non-isometry
   }
   for (const auto& t : subframe_poses_)
   {
     ASSERT_ISOMETRY(t.second)  // unsanitized input, could contain a non-isometry
   }
-  global_collision_body_transforms_.resize(attach_trans.size());
+
+  // TODO(felixvd): These are initialized as identity because the parent link transform is not
+  //                known to the AttachedBody. Is computeTransform called before they are used?
+  //                Otherwise they will not be correct.
```

Comment:
I think this should become an issue, if this gets merged, so we can track down potential future bugs more easily.

Reply:
If no one is sure, then definitely. Otherwise, tracking "maybe-bugs" will probably just add noise, and this comment and some documentation in the existing issues/PRs may be sufficient.

Reply:
`computeTransform` is called in `attachBody` so all the attached bodies in a `robot_state` are safe. I grepped through the codebase, and everywhere where `AttachedBody` objects are handled, it's either to prepare a call to `attachBody` (safe) or to get an AttachedBody from a robot state (also safe). I'll leave this discussion here for now, but I'm pretty confident that this is fine.